### DISCUSSION TITLE
Simpler way to get a user's policies needed

### DISCUSSION THREAD

**Discussion by stx-chris** [directus/directus#24691] [Feature Requests]
### Describe the Bug

Although technically not a bug, using policies in actual application turns into a cumbersome endeavour. It ought to easy to get the names of all policies assigned to a user, including the ones attached to their role. (Using policy ids instead of names in an app is not a practical approach)

If I want to know which policies a user belongs to, I have to execute the following steps:
1. Set up a policy and add `Read` permission to at least `directus_access`, `directus_policies`,`directus_users`
2. Restrict access to these tables to the user and the user's role respectively
3. Retrieve the user's record including the filters `policy.policies.name`, `role.policies.policy.name` to get the actual policy names
4. Merge the policy names from their different levels in the returning user object into a single array to be usable for lookups within an application

Can we have a helper function within the SDK, maybe `getUserPolicies(user)` that returns an array of all policies a user and their role belongs to? Alternatively, an SDK function that returns a boolean whether the user has a certain role.

### To Reproduce

Set up user, role and policy and retrieve the user's info via `/users/me`

### Directus Version

v11.0.0

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

PostgreSQL

**Comment by rijkvanzanten** [comment]
We currently do have a `/permissions/me`. Feels appropriate to add a `/policies/me` as well! @hanneskuettner thoughts?

**Comment by hanneskuettner** [comment]
Let's do it! Any thoughts on bringing this to the service level as well, so it is available for API extensions? I guess this falls into a similar vain as https://github.com/directus/directus/issues/23318

**Comment by rijkvanzanten** [comment]
We don't currently do that for `/permissions/me`, but I think we should! Two birds one stone

**Comment by NilsBaumgartner1994** [comment]
Please also add a simple way to get it in the sdk too: const result = await client.request(readUserPolicies());

**Comment by flipswitchingmonkey** [comment]
For some reason when a user is Admin, the user object returned by `/users/me` already contains an array of `policies`, however this is not a case for other users. This would solve a lot of problems since I'm always getting that after login anyway.

I think what's needed is to change the "App Access" permission defaults to include the policy permissions as above, similar to how it is already done for roles etc.

Edit: my current workaround is to create a "Default User Policy" that uses the App Access default and adds `directus_access` with `read` and `user _eq $CURRENT_USER` rule and `directus_policies` `read` with `id _in $CURRENT_POLICIES`. Also disable the `users` field.

Edit-Edit: actually I think `id _in $CURRENT_POLICIES` does not work :(

**Comment by flipswitchingmonkey** [comment]
And the trouble continues - the `policies` array actually contains the `id` of the `directus_access` entry, not the `id of the `directus_policies` entry... :(

**Comment by hanneskuettner** [comment]
> Edit-Edit: actually I think id _in $CURRENT_POLICIES does not work :(

That seems to be indeed a bug of the current version, see #23695. It should work with `id _in $CURRENT_POLICIES.id` tho. 

See #23728 for a PR fixing the bug

**Comment by isawk** [comment]
> > Edit-Edit: actually I think id _in $CURRENT_POLICIES does not work :(
> 
> 
> 
> That seems to be indeed a bug of the current version, see #23695. It should work with `id _in $CURRENT_POLICIES.id` tho. 
> 
> 
> 
> See #23728 for a PR fixing the bug

Would be ideal to also have ability when setting collection rules to have a _in ability for use cases where by a user on multi tenant system has access to more than one tenant.

{
  tenant.id IN $CURRENT_USER.tenants.tenant.id 

} 

I know there is "one of" , but that is not dynamic

**Comment by brad-datazo** [comment]
> And the trouble continues - the `policies` array actually contains the `id` of the `directus_access` entry, not the `id of the `directus_policies` entry... :(

Is this a separate bug? I can't find it logged anywhere.

I'm also running into this when trying to retrieve the names of the policies of the logged in user. The sdk indicates that a `directus_user` should have an array of `policies` but to actually get the name and id of the policy you must query for it an extra layer deep (`policies.policy.name`) it like this:

```ts
directusClientInstance.request(
      readMe({
        fields: [
          '*',
          {
            policies: ['*', { policy: ['name', 'id'] }],
            role: ['*', { policies: [{ policy: ['name', 'id'] }] }],
          },
        ],
      }),
    );
```

... that query actually does work (as long as the logged in user has proper permissions) but will give you a typescript syntax error
```
Object literal may only specify known properties, and 'policy' does not exist in type '{ users?: QueryFields<MySchema, { id: string; first_name: string | null; last_name: string | null; email: string | null; password: string | null; location: string | null; title: string | null; ... 20 more ...; policies: string[] | { ...; }[]; }[]> | undefined; permissions?: QueryFields<...> | undefined; roles?: Q...
```

According to the sdk the correct query is
``` ts
directusClientInstance.request(
      readMe({
        fields: [
          '*',
          {
            policies: ['id', 'name'],
            role: ['*', { policies: ['id', 'name'] }],
          },
        ],
      }),
    );
```
but it will not give you the policy names or ids. Weirdly it only gives me back the `id` of the logged in user.

Sorry if that was a mess - my first contribution to an open source project. Thanks for your work!

**Comment by rijkvanzanten** [comment]
@brad-datazo I think your confusion stems from the fact that policies are attached to users / roles through a junction table. The `policies` field on a user points to the `directus_access` table under the hood, which then has a relationship to the attached policy on the other end. In your SDK query example you're accessing `id` and `name` on the junction table rather than the policies table.
